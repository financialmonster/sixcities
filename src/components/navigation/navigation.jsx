import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import {bindActionCreators} from 'redux';

import {activeCityChange} from '../../actions.js';
import {citiesNames} from '../../selectors.js';

const Navigation = ({activeCity, navItemClickHandler, citiesNames}) => {
    return (
        <React.Fragment>
            <h1 className="visually-hidden">Cities</h1>
            <div className="tabs">
                <section className="locations container">
                    <ul className="locations__list tabs__list">
                        {
                            citiesNames.map((item) => {
                                const className = (item === activeCity)
                                    ? `locations__item-link tabs__item tabs__item--active`
                                    : `locations__item-link tabs__item`;

                                return (
                                    <li key={item} className="locations__item">
                                        <a className={className} href="#" id={item}
                                            onClick={() => navItemClickHandler(item)}
                                        >
                                            <span>{item}</span>
                                        </a>
                                    </li>
                                );
                            })
                        }
                    </ul>
                </section>
            </div>
        </React.Fragment>
    )
}

Navigation.propTypes = {
    activeCity: PropTypes.string,
    navItemClickHandler: PropTypes.func.isRequired,
    citiesNames: PropTypes.arrayOf(PropTypes.string)
}

const mapStateToProps = (state) => {
    return {
        activeCity: state.activeCity,
        citiesNames: citiesNames(state)
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        navItemClickHandler: bindActionCreators(activeCityChange, dispatch)
    }
}

export {Navigation};
export default connect(mapStateToProps, mapDispatchToProps)(Navigation);
