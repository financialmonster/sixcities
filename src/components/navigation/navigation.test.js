import React from 'react';
import renderer from 'react-test-renderer';

import {Navigation} from './navigation.jsx';

const citiesNames = [`Paris`, `Amsterdam`, `Tokio`, `Kiev`, `Moscow`, `Krakow`];

it(`Correctly renders the Navigation component`, () => {
    const tree = renderer.create(
        <Navigation navItemClickHandler={() => {}} activeCity="Paris"
            citiesNames={citiesNames}
        />)
    .toJSON();

    expect(tree).toMatchSnapshot();
});
