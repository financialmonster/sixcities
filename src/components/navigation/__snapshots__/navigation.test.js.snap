// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders the Navigation component 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>,
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="#"
            id="Paris"
            onClick={[Function]}
            title=""
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            id="Amsterdam"
            onClick={[Function]}
            title="To the offers of Amsterdam"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            id="Tokio"
            onClick={[Function]}
            title="To the offers of Tokio"
          >
            <span>
              Tokio
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            id="Kiev"
            onClick={[Function]}
            title="To the offers of Kiev"
          >
            <span>
              Kiev
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            id="Moscow"
            onClick={[Function]}
            title="To the offers of Moscow"
          >
            <span>
              Moscow
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            id="Krakow"
            onClick={[Function]}
            title="To the offers of Krakow"
          >
            <span>
              Krakow
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>,
]
`;
